PayloadPkg/FirmwareUpdate/FirmwareUpdateHelper.c

```diff
@@ -1151,3 +1160,58 @@ Reboot (
   ResetSystem (ResetType);
   CpuDeadLoop ();
 }
+
+/**
+  Verify the firmware internal structure.
+
+  @param[in] ImageHdr     Pointer to the fw mgmt capsule image header
+
+  @retval  EFI_SUCCESS    The operation completed successfully.
+  @retval  others         There is error happening.
+**/
+EFI_STATUS
+VerifyFwStruct (
+  IN  EFI_FW_MGMT_CAP_IMAGE_HEADER  *ImageHdr
+  )
+{
+  UINTN                 ImageBase;
+  UINTN                 ImageOffset;
+  CPU_MICROCODE_HEADER  *UCodeHdr;
+  UINT8                 *ImageByte;
+
+  if ((UINT32)ImageHdr->UpdateHardwareInstance == FLASH_MAP_SIG_UCODE) {
+
```

Comment:
Could you move the ucode slot size checks to a different method?  VerifyFwStruct can be used for other components also.